File: Reactfrontend/src/services/categoryService.js
import api from "./api";

const getAllCategories = () => {
  return api.get("/categories");
};

const getCategoryById = (id) => {
  return api.get(`/categories/${id}`);
};

const createCategory = (category) => {
  return api.post("/categories", category);
};

const updateCategory = (id, category) => {
  return api.put(`/categories/${id}`, category);
};

const deleteCategory = (id) => {
  return api.delete(`/categories/${id}`);
};

const categoryService = {
  getAllCategories,
  getCategoryById,
  createCategory,
  updateCategory,
  deleteCategory,
};

export {
  getAllCategories,
  getCategoryById,
  createCategory,
  updateCategory,
  deleteCategory,
};
export const getCategories = () => {
  return api.get("/categories");
};

export default categoryService;

File: backend/src/main/java/com/smartcart/service/CategoryService.java
package com.smartcart.service;

import com.smartcart.entity.Category;
import com.smartcart.repository.CategoryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CategoryService {

    @Autowired
    private CategoryRepository categoryRepository;

    // Add Category
    public Category saveCategory(Category category) {
        return categoryRepository.save(category);
    }

    // Get All Categories
    public List<Category> getAllCategories() {
        return categoryRepository.findAll();
    }

    // Get Category By Id
    public Optional<Category> getCategoryById(Long id) {
        return categoryRepository.findById(id);
    }

    // Update Category
    public Category updateCategory(Long id, Category category) {

        Category existingCategory = categoryRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Category not found"));

        existingCategory.setName(category.getName());
        existingCategory.setDescription(category.getDescription());

        return categoryRepository.save(existingCategory);
    }

    // Delete Category
    public void deleteCategory(Long id) {
        categoryRepository.deleteById(id);
    }
}